Source/Core/DiscIO/RiivolutionParser.cpp

```diff
@@ -256,8 +271,18 @@ bool Disc::IsValidForGame(const std::string& game_id, std::optional<u16> revisio
   return true;
 }
 
-std::vector<Patch> Disc::GeneratePatches(const std::string& game_id) const
+std::vector<Patch> Disc::GeneratePatches(const std::string& game_id, u32 ng_id) const
 {
+  IOS::HLE::Kernel m_ios;
+  if (ng_id == 0)
+  {
+    // If the patch did not include a hard-coded device ID, use the one from the emulated console.
+    ng_id = m_ios.GetIOSC().GetDeviceId();
+  }
```

Comment:
This isn't a member variable, and as far as I can tell you only use it inside the `if`. You should probably move it inside:
```suggestion
  if (ng_id == 0)
  {
    // If the patch did not include a hard-coded device ID, use the one from the emulated console.
    IOS::HLE::Kernel ios;
    ng_id = ios.GetIOSC().GetDeviceId();
  }
```

Not sure if theres a better way to get the device id though.

Reply:
Yeah, that code's a bit messy because I was testing how I can get the device ID. Once I (or someone else) figure(s) out why the current approach crashes and how to fix it, I'll clean that up.